admin-notice/js/script.js

```diff
@@ -0,0 +1,72 @@
+const vipAdminNoticeDataAttribute = 'data-vip-admin-notice';
+const vipAdminNoticeCookieName = 'vip-admin-notice-dismissed'
+const vipAdminNoticeCookieDelimeter = '|';
+
+function tryGetNoticeContainer(currentElement) {
+    if (currentElement === null || currentElement.hasAttribute(vipAdminNoticeDataAttribute)) {
+        return currentElement;
+    }
+    return tryGetNoticeContainer(currentElement.parentElement);
+}
+
+function getCookie() {
+    const value = `; ${document.cookie}`;
+    const parts = value.split(`; ${vipAdminNoticeCookieName}=`);
+    if (parts.length === 2) return parts.pop().split(';').shift();
+}
+
+function getDismissedNotices() {
+    const cookie_value = getCookie() || '';
+    return cookie_value.split(vipAdminNoticeCookieDelimeter);
+
+}
+
+function emitDismissedCookie(containerElement) {
+    const dismissIdentifier = containerElement?.getAttribute(vipAdminNoticeDataAttribute);
+    if (dismissIdentifier) {
+        const previousCookieValue = getCookie();
+        const cookieValue = previousCookieValue ? `${previousCookieValue}${vipAdminNoticeCookieDelimeter}${dismissIdentifier}` : dismissIdentifier;
+
+        let expiryDate = new Date();
+        expiryDate.setFullYear(expiryDate.getFullYear() + 1)
+        document.cookie = `${vipAdminNoticeCookieName}=${cookieValue};expires=${expiryDate.toUTCString()};path = /`;
+    }
+}
+
+function hideNotice(notice) {
+    notice.style.display = 'none';
+}
+
+function onDismissClicked(event) {
+    const noticeContainer = tryGetNoticeContainer(event?.target);
+    if (noticeContainer) {
+        hideNotice(noticeContainer);
+        emitDismissedCookie(noticeContainer);
+    }
+}
+
+function registerDismissHooks() {
+    const dismissButtons = document.getElementsByClassName("vip-notice-dismiss");
+    for (const button of dismissButtons) {
+        button.onclick = onDismissClicked;
+    }
+}
+
+function hideDismissedNotices() {
```

Comment:
This was originally in server-side php code, but our standard warn me against using cookies on server-side due to caching. So I moved it here

Reply:
This code will only be shown when logged in, and requests with logged-in cookies get a `PASS` at the edge, so there wouldn't be any page cache involved here.

Reply:
Oh I see, Is there a comment I should add to pass linting then? Is that a common practice?

Reply:
BTW what is your preference? I kinda like that both setting and reading cookie logic is in Javascript in the same file as opposed to having it split between the backend and frontend and there is no need to avoid linting issues. On the other hand, having notice to pop-up before js starts up to hide is less optimal.

Reply:
If someone dismisses a notice, and then loads when JS is not available, does the current solution means that it would show up again for them?

Is there a user meta / site meta field that is set which would avoid a notice from displaying from the back-end? i.e. store the fact a notice is dismissed in the DB, and not in a cookie (which could be wiped at any time).

How does WP core, or other popular plugins (Yoast SEO?) handle dismissible notices?

Reply:
I couldn't find a standard way to do this. [Some plugin](https://github.com/Automattic/vip-go-mu-plugins/blob/2f4d90e41975b252425d85e2fd7e5b518c83fb5f/performance/bulk-edit.php#L82) seems to be leveraging `dismissed_wp_pointers` user meta (it has ajax action that can be used to add dismissed pointers). Seems nice but a bit hacky.

Some articles I read mentioned cookies being used in most cases which is in line with the request for this change: `Dismissable (persistently...can be cookie based)` so I went with cookie.